Automattic/wp-calypso
# Add triennial plans support in calypso-products and MVP implementation in checkout

#### Proposed Changes
Note: started this, but WIP.

* Add selectors, constants, and utils to support triennial (3 year) plans and billing intervals in calypso-products.
* End goal is to be able to go to the triennial plans checkout slug, and add a triennial plan (Personal, Premium, Business, or Ecommerce) to the cart. Pending API changes. The follow-up with calls to action will be handled in a separate PR.

#### To do:
- [x] Real IDs (currently just a placeholder)
- [ ] Check upgrades, what can upgrade to what in 3 y plans.
- [x] String freeze for new strings
- [x] More tests + make tests in web pass.

#### Testing Instructions
* Automated tests pass.
* Spin up this branch locally (or via live link below)
* ~~Sandbox D96286-code~~ (merged 20220106)
* Sandbox store. 
* Go to calypso.localhost:3000/checkout/thisisatestforsomestuff.wordpress.com/ecommerce-bundle-3y 
<img width="1145" alt="Screenshot 2022-12-22 at 15 25 49" src="https://user-images.githubusercontent.com/52675688/209220094-c5c82a48-ec64-4c50-abb5-42372c389ff8.png">

- Purchase a plan. Also go ahead and test other plans with 3 year intervals (i.e. Business, Premium, Personal).
- The purchase should work, and you'll get to the onboarding flow.

- Make sure that there's no "upsell" to a biennial plan from a triennial, and no other questionable upsells to lower plans.
- If you go through the upgrade or checkout flow with a yearly plan, you should be able to upgrade to a 2y or 3y plan on checkout.
- If you go to checkout for a 2-y plan (e.g. via direct checkout link), you should see the dropdown with the option to upgrade to a 3y plan.

- Following the signup/checkout, once you are at the dashboard, have a look around and make sure that the 3-year plans work as expected, e.g. go to the plans page, "my plan", site settings, etc.

#### Pre-merge Checklist





Related to #
